### DISCUSSION TITLE
Ability to transform the id when importing from SQL

### DISCUSSION THREAD

**Discussion by loraderon** [ravendb/ravendb#17479] [Ideas]
I really want to control the document id when importing data.
I tried with both `this.Id = "Foo/" + this.Bar + "/" + this.Baz` and `this["@id"] = "Foo/" + this.Bar + "/" + this.Baz` but these just added properties to the document instead of chaning the actual id.

**Comment by ayende** [comment]
There is no support for that, I'm afraid.
The document id is computed from the table primary key(s) only

**Reply by loraderon** [reply]
I noticed this and it would have worked for me if the `date` column from postgresql was formatted to `yyyy-MM-dd` instead of `yyyy-MM-dd 00:00:00`
